integrations/SnapPixel/util.js

```diff
@@ -153,14 +157,19 @@ const ecommEventPayload = (event, message) => {
       }
       payload = {
         ...payload,
-        transaction_id: get(message, "properties.transaction_id"),
         item_ids: itemIds,
       };
       break;
     }
     case "product viewed": {
-      const itemIds = [];
-      itemIds.push(get(message, "properties.product_id"));
+      let itemIds = [];
+      const pId = get(message, "properties.product_id");
+      if (isDefinedAndNotNull(pId)) {
```

Comment:
Unless the product ID can be `0`, I'd suggest we use the below,
```suggestion
      if (pId) {
```

Reply:
Done